### PR TITLE
Some small changes

### PR DESCRIPTION
**Pull request by julian-smith-artifex-com**

### CODE REVIEW

**Review by JorjMcKie — APPROVED** [review]

File: src/__init__.py [hunk 1]
```diff
@@ -2614,7 +2615,10 @@ def __exit__(self, *args):
         self.close()
 
     def __getitem__(self, i: int =0):
-        assert isinstance(i, int) or (isinstance(i, tuple) and len(i) == 2 and all(isinstance(x, int) for x in i))
+        if isinstance(i, slice):
+            return [self[j] for j in range(*i.indices(len(self)))]
+        assert isinstance(i, int) or (isinstance(i, tuple) and len(i) == 2 and all(isinstance(x, int) for x in i)), \
+                f'Invalid item number: {i=}.'
```
Comment (JorjMcKie) [inline src/__init__.py:2621]: Can we switch to consistently using double apostrophes " in changed code.

File: src/__init__.py [hunk 2]
```diff
@@ -15644,6 +15632,13 @@ def JM_get_annot_xref_list( page_obj):
     return names
 
 
+def JM_get_annot_xref_list2(page):
+    page = page._pdf_page()
+    if not page.m_internal:
+        return list()
+    return JM_get_annot_xref_list( page.obj())
```
Comment (JorjMcKie) [inline src/__init__.py:15639]: Can we stop using spaces after opening brackets like `( page.obj())`.

File: tests/test_general.py [hunk 3]
```diff
@@ -1178,3 +1178,17 @@ def check(filename=None, stream=None, filetype=None, exception=None):
...
+        return
+    path = os.path.abspath(f'{__file__}/../../tests/resources/2.pdf')
+    doc = fitz.open(path)
+    print()
+    for p in doc:
+        print(f'for p in doc: {p=}.')
+    for p in list(doc)[:]:
+        print(f'for p in list(doc)[:]: {p=}.')
+    for p in doc[:]:
+        print(f'for p in doc[:]: {p=}.')
```
Comment (JorjMcKie) [inline tests/test_general.py:1194]: Can we stop using single apostrophes in changed code.

**Comment by julian-smith-artifex-com** [comment]
Merging this despite test_quick failure, because the failure is caused by changes to mupdf, and i have further commits to address this.

**Pull request merged**